kernel/fs.c

```diff
@@ -695,3 +703,173 @@ nameiparent(char *path, char *name)
 {
   return namex(path, 1, name);
 }
+
+
+void swap_in_page(uint va) {
+  struct page_info *page = find_page_info(va);
+  if (!page->in_swap) {
```

Comment:
No NULL pointer check.
In fact a swapped out page should have `find_page_info` `return NULL;`.